src/promptflow/promptflow/executor/flow_validator.py

```diff
@@ -126,10 +128,27 @@ def resolve_flow_inputs_type(flow: Flow, inputs: Mapping[str, Any], idx: Optiona
             try:
                 if k in inputs:
                     updated_inputs[k] = v.type.parse(inputs[k])
+            except JSONDecodeError as e:
+                line_info = "" if idx is None else f" in line {idx} of input data"
+                msg_format = (
+                    "Failed to parse the flow input. The value for flow input '{flow_input_name}'{line_info} "
```

Comment:
Let's extract `'{flow_input_name}'{line_info}` part into an argument (maybe named `{flow_input_info}`?) to make it better to understand.